src/items/repositories/item.repository.ts

```diff
@@ -84,6 +84,18 @@ export class ItemRepository extends Repository<Item> {
       .orderBy('item.createdAt', 'DESC')
       .getOne();
   }
+  async listAnItem(itemId: string, price: number): Promise<Item> {
+    await this.update(
+      {
+        itemId,
+      },
+      { price: String(price), status: 1 }
+    );
+
+    return await this.findOne({
+      where: { itemId },
+    });
+  }
 
   async createItem(itemRequestDto: ItemRequestDto, accountId: string): Promise<Item> {
     const { tokenId, name, collectionAddress, description, royalty, status, image, wei } =
```

Comment:
Please make sure the `wei` in the dto has validation rules (the ones that comes to my mind are not empty, numeric and not negative)

Reply:
What do you mean with "wei"?